nautobot/docs/development/core/kubernetes-job-support.md

```diff
@@ -0,0 +1,509 @@
+pod/celery-beat-6fb67477b7-rsw62   1/1     Running   0             30m
+pod/db-8687b48964-gtvtc            1/1     Running   0             30m
+pod/nautobot-679bdc765-pl2ld       1/1     Running   0             30m
+pod/redis-7cc58577c-tl5sq          1/1     Running   0             30m
+
+NAME                 TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
+service/db           ClusterIP   10.111.0.30     <none>        5432/TCP   30m
+service/kubernetes   ClusterIP   10.96.0.1       <none>        443/TCP    30m
+service/nautobot     ClusterIP   10.106.32.53    <none>        8080/TCP   30m
+service/redis        ClusterIP   10.102.99.143   <none>        6379/TCP   30m
+
+NAME                          READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/celery-beat   1/1     1            1           30m
+deployment.apps/db            1/1     1            1           30m
+deployment.apps/nautobot      1/1     1            1           30m
+deployment.apps/redis         1/1     1            1           30m
+```
+
+## Environment Variables and Job Pod Manifest
+
+## Running Jobs
+
+To pass stdin into the nautobot container that exists with in the nautobot pod, first you need to get the name of the nautobot pod.
+
+You can execute the following command to see all the pods.
+
+```bash
+kubectl get pods
+```
+
+You should see the following output and note that the pod with the prefix `nautobot-*`. That is what we need for our command.
+
+```bash
+NAME                               READY   STATUS    RESTARTS      AGE
+pod/celery-beat-6fb67477b7-rsw62   1/1     Running   0             30m
+pod/db-8687b48964-gtvtc            1/1     Running   0             30m
+pod/nautobot-679bdc765-pl2ld       1/1     Running   0             30m
+pod/redis-7cc58577c-tl5sq          1/1     Running   0             30m
+```
+
+Now to get to the CLI of the nautobot container in the pod, we execute the following command:
+
+```bash
+kubectl exec --stdin --tty <nautobot-pod-name> -- /bin/bash
+```
+
+You should see that we are inside the container as the root user:
+
+```bash
+root@nautobot-679bdc765-pl2ld:/source#
+```
+
+### Create and Assign a Kubernetes Job Queue to a Job
+
+Now we are inside the container, we can access `nbshell` using `nautobot-server nbshell` and make modifications to our database records. You should see the following output:
+
+```py
+# Django version 4.2.16
+# Nautobot version 2.4.0a1
+# Example Nautobot App version 1.0.0
+Python 3.11.10 (main, Nov 12 2024, 22:19:37) [GCC 12.2.0] on linux
+Type "help", "copyright", "credits" or "license" for more information.
+(InteractiveConsole)
+>>>
+```
+
+In order to run kubernetes jobs, we need to create a [Job Queue](../../user-guide/platform-functionality/jobs/jobqueue.md) with type Kubernetes.
+
+```py
+>>> from nautobot.extras.choices import JobQueueTypeChoices
+>>> jq = JobQueue.objects.create(name="kubernetes", queue_type=JobQueueTypeChoices.TYPE_KUBERNETES)
```

Comment:
It would be nice to have a possibility to directly copy the example code and paste it into a console without a need to remove the `>>>` signs. Maybe add code without it and screenshot of command run in your terminal? :thinking: 

The same goes to the code below with `Job` creating code and `curl` command - but there it's even more important because there is more to copy :D

Reply:
Moved these to UI screenshots